src/components/notifications/Message.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * Displays a notification message with support for navigating to another page,
+ * or some other action, when certain message types are clicked / tapped.
+ *
+ * @author psarando
+ */
+import React from "react";
+
+import Link from "next/link";
+
+import { getDisplayMessage } from "./utils";
+
+import NotificationCategory from "components/models/NotificationCategory";
+import SystemId from "components/models/systemId";
+
+import { getAnalysisDetailsLinkRefs } from "components/analyses/utils";
+import { getListingPath } from "components/apps/utils";
+import { getFolderPage, getParentPath } from "components/data/utils";
+import NameLink from "components/utils/NameLink";
+
+import { Typography } from "@material-ui/core";
+
+function MessageLink(props) {
+    const { message, href, as } = props;
+
+    return (
+        <Link href={href} as={as} passHref>
+            <NameLink name={message} />
```

Comment:
Now that we are using `NameLink` for other things as well should this be renamed to something link `DELink` ?